src/pages/Scan.vue

```diff
@@ -92,25 +92,42 @@ export default {
                     // Success function
                     function(response) {
                         // Add error message for invalid and fail codes.
-                        if (
-                            response.data.invalid.length > 0
-                        ) {
-                            this.showFail();
-                            this.errorMessage = "Please enter a valid voucher code.";
-                        } else if (
-                            response.data.fail.length > 0
-                        ) {
+                        var data = response.data;
+
+                        if (data.invalid.length + data.fail.length === 1) {
+                            // single mismatch handler;
+                            if (data.invalid.length > 0) {
+                                this.showFail();
+                                this.errorMessage = "Please enter a valid voucher code.";
+
+                            } else if (data.fail.length > 0) {
+                                this.showFail();
+                                this.errorMessage = "That voucher may have been used already.";
+                            }
+
+                        } else if (data.invalid.length + data.fail.length > 1) {
+                            // rough multifailure manager
                             this.showFail();
-                            this.errorMessage = "That voucher may have been used already.";
+                            this.errorMessage = "[xXx] " +
+                                data.success.length +
+                                " accepted, " +
+                                data.fail.length +
+                                " rejected and " +
+                                data.invalid.length +
+                                " were invalid.";
```

Comment:
It's so you can easily see where the start and end of a multi-line is more easily.

It also helps with git diffs when stuff is added to the end so it doesn't appear that 2 lines have changed.

Same theory as }); on a new line